src/server/transaction/server/driver.go

```diff
@@ -225,60 +221,81 @@ func (d *driver) appendTransaction(
 	txn *transaction.Transaction,
 	items []*transaction.TransactionRequest,
 ) (*transaction.TransactionInfo, error) {
-	// Run this thing in a loop in case we get a conflict, time out after some tries
-	for i := 0; i < 10; i++ {
-		// We first do a dryrun of the transaction to
+	return d.updateTransaction(ctx, false, txn, func(txnCtx *txncontext.TransactionContext, info *transaction.TransactionInfo) (*transaction.TransactionInfo, error) {
+		// We do a dryrun of the transaction to
 		// 1. make sure the appended request is valid
 		// 2. Capture the result of the request to be returned
-		var numRequests, numResponses int
-		var newResponses []*transaction.TransactionResponse
-
-		if err := d.txnEnv.WithReadContext(ctx, func(txnCtx *txncontext.TransactionContext) error {
-			// Get the existing transaction and append the new requests
-			info := &transaction.TransactionInfo{}
-			err := d.transactions.ReadWrite(txnCtx.SqlTx).Get(txn.ID, info)
-			if err != nil {
-				return err
-			}
-
-			// Save the length so that we can check that nothing else modifies the
-			// transaction in the meantime
-			numRequests = len(info.Requests)
-			numResponses = len(info.Responses)
-			info.Requests = append(info.Requests, items...)
-			if newResponses != nil {
-				info.Responses = newResponses
-			}
+		info.Requests = append(info.Requests, items...)
```

Comment:
This will have different behavior from before, since now a `CreatePipeline` retry will count against the 10 attempts, rather than happening entirely inside of the SQL transaction. That case *should* be rare - honestly I'm not sure the real world cases where we expect most of this code to be exercised, anyway